lib/exabgp/rib/store.py

```diff
@@ -182,6 +182,14 @@ def insert_announced (self, change, force=False):
 				if self.cache and change_nlri_index not in self._seen.get(change.nlri.family(),{}):
 					return
 
+		if not force and self.cache:
+			announced = self._seen.get(change.nlri.family(),{})
+			if change_nlri_index in announced and change.nlri.action == OUT.ANNOUNCE:
+				old_change = announced[change_nlri_index]
+				# it is a duplicate route
+				if old_change.attributes.index() == change.attributes.index() and old_change.nlri.nexthop.index() == change.nlri.nexthop.index():
```

Comment:
The route in announced could be a withdrawal, if it is a withdrawal, then we need to just replace it with this new announce.
So I guess the test should be updated to:
```if old_change.nlri.action == OUT.ANNOUNCE  and old_change.attributes.index() == change.attributes.index() and old_change.nlri.nexthop.index() == change.nlri.nexthop.index():
```
what do you think ?